device/pcie/pci_device.hpp

```diff
@@ -34,84 +34,144 @@ struct PciDeviceInfo
+    int get_pci_device_id() const { return info.device_id; }
+
+    /**
+     * @return PCI revision value from sysfs.
+     * TODO: target for removal; upper layers should not care about this.
+     */
+    int get_pci_revision() const { return revision; }
+
+    /**
+     * @return what architecture this device is (e.g. Wormhole, Blackhole, etc.)
+     */
+    tt::ARCH get_arch() const { return arch; }
+
+    // Note: byte_addr is (mostly but not always) offset into BAR0.  This
+    // interface assumes the caller knows what they are doing - but it's unclear
+    // how to use this interface correctly without knowing details of the chip
+    // and its state.
+    // TODO: build a proper abstraction for IO.  At this level, that is access
+    // to registers in BAR0 (although possibly the right abstraction is to add
+    // methods that perform specific operations as opposed to generic register
+    // read/write methods) and access to segments of BAR0/4 that are mapped to
+    // NOC endpoints.  Probably worth waiting for the KMD to start owning the
+    // resource management aspect of these PCIe->NOC mappings (the "TLBs")
+    // before doing too much work here...
     void write_block(uint64_t byte_addr, uint64_t num_bytes, const uint8_t* buffer_addr);
     void read_block(uint64_t byte_addr, uint64_t num_bytes, uint8_t* buffer_addr);
     void write_regs(uint32_t byte_addr, uint32_t word_len, const void *data);
     void write_regs(volatile uint32_t *dest, const uint32_t *src, uint32_t word_len);
     void read_regs(uint32_t byte_addr, uint32_t word_len, void *data);
     void write_tlb_reg(uint32_t byte_addr, std::uint64_t value_lower, std::uint64_t value_upper, std::uint32_t tlb_cfg_reg_size);
 
-    void open_hugepage_per_host_mem_ch(uint32_t num_host_mem_channels);
     tt::umd::architecture_implementation* get_architecture_implementation() const { return architecture_implementation.get(); }
 
-    PciDeviceInfo info;
-
-    int device_id;  // N in /dev/tenstorrent/N
-    int logical_id; // TODO: does not belong in here
-    int device_fd = -1;
-
-    // PCIe device info
-    int numa_node;
-    std::uint32_t pcie_device_id;
-    std::uint32_t pcie_revision_id;
-
-    // BAR and regs mapping setup
-    std::vector<int> device_fd_per_host_ch;
+public:
+    // TODO: we can and should make all of these private.
     void *bar0_uc = nullptr;
-    std::size_t bar0_uc_size = 0;
-    std::size_t bar0_uc_offset = 0;
+    size_t bar0_uc_size = 0;
+    size_t bar0_uc_offset = 0;
 
     void *bar0_wc = nullptr;
-    std::size_t bar0_wc_size = 0;
+    size_t bar0_wc_size = 0;
 
     void *bar2_uc = nullptr;
-    std::size_t bar2_uc_size;
+    size_t bar2_uc_size;
 
     void *bar4_wc = nullptr;
-    std::uint64_t bar4_wc_size;
+    uint64_t bar4_wc_size;
 
+    // TODO: let's get rid of this unless we need to run UMD on WH systems with
+    // shrunk BAR0.  If we don't (and we shouldn't), then we can just use BAR0
+    // and simplify the code.
```

Comment:
Right, I was thinking the same, this system_reg_mapping thing is a bit confusing. We will end up refactoring it. First it needs to be raised to TTDevice class since it is arch specific.

No action on this comment.

Reply:
I just confirmed with syseng that future systems will not have a shrunken BAR0, even with large numbers of chips.

Reply:
#104